### PR TITLE
Adding API gateway configuration get API into core REST API 

### PR DESCRIPTION
**Pull request by tgtshanika**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

- Adding API gateway configuration get API to core REST API
- Updating swagger definition of core REST APIs

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] ** Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] ** Are all UI and API inputs run through forms or serializers?**
- [ ] ** Are all external inputs validated and sanitized appropriately?**
- [ ] ** Does all branching logic have a default case?**
- [ ] ** Does this solution handle outliers and edge cases gracefully?**
- [ ] ** Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/ApisApi.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.wso2.carbon.apimgt.rest.api.core.dto.ErrorDTO;
+
+import org.wso2.msf4j.Microservice;
+import org.wso2.msf4j.Request;
+import org.osgi.service.component.annotations.Component;
+
+import java.io.InputStream;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.*;
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/ApisApi.java]: Remove the * import

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIMgtAdminService.java [hunk 2]
```diff
@@ -122,4 +122,21 @@
...
+     * Retrieve API's gateway configuration
+     *
+     * @param apiId     UUID of the API
+     * @return          The API gateway configuration
+     * @throws APIManagementException   If failed to retrive API gateway config
+     */
+    String getAPIGatewayServiceConfig(String apiId) throws APIManagementException;
+
+    /**
+     *
```
Comment (sajhak) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIMgtAdminService.java]: Description is missing

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/ApisApiServiceFactory.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.wso2.carbon.apimgt.rest.api.core.factories;
+
+import org.wso2.carbon.apimgt.rest.api.core.ApisApiService;
+import org.wso2.carbon.apimgt.rest.api.core.impl.ApisApiServiceImpl;
+
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/ApisApiServiceFactory.java]: Add method comments

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImpl.java [hunk 4]
```diff
@@ -170,4 +170,35 @@ public void registerGatewayLabels(List<Label> labels, String overwriteLabels) th
         }
 
     }
+
+    @Override
+    public String getAPIGatewayServiceConfig(String apiId) throws APIManagementException {
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImpl.java]: Shall we add the method comments using @see?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/ApisApiServiceFactory.java (hunk 3 above)
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/ApisApiServiceFactory.java]: This is an autogenerated class

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIMgtAdminService.java (hunk 2 above)
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIMgtAdminService.java]: Shall we please use a more specific exception which extends from APIManagementException?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/ApisSummaryApi.java [hunk 5]
```diff
@@ -44,8 +42,11 @@
         @io.swagger.annotations.ApiResponse(code = 404, message = "Not Found. The resource to be updated does not exist. ", response = APISummaryListDTO.class),
         
         @io.swagger.annotations.ApiResponse(code = 406, message = "Not Acceptable. The requested media type is not supported ", response = APISummaryListDTO.class) })
-    public Response apisSummaryGet()
+    public Response apisSummaryGet(@ApiParam(value = "Media types acceptable for the response. Default is application/json. " , defaultValue="application/json")@HeaderParam("Accept") String accept
+,@ApiParam(value = "Validator for conditional requests; based on the ETag of the formerly retrieved variant of the resourec. " )@HeaderParam("If-None-Match") String ifNoneMatch
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/ApisSummaryApi.java]: Do we really need ETags for this resource/api? I don't think so.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/impl/ApisApiServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            }
+            
+            apiGatewayConfig = apiMgtAdminService.getAPIGatewayServiceConfig(apiId);
+
+            if (apiGatewayConfig != null) {
+                return Response.ok().header(HttpHeaders.ETAG, "\"" + existingFingerprint + "\"").entity(apiGatewayConfig)
+                        .build();
+            } else {
+                String msg = "API is not found with apiId : " + apiId;
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(msg, 900314L, msg);
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/impl/ApisApiServiceImpl.java]: Shall we use the ExceptionCodes concept in here as well?

**Pull request merged**